### DISCUSSION TITLE
Keep local yaml modifications

### DISCUSSION THREAD

**Discussion by blamaz** [make-all/tuya-local#501] [General]
Hi,
I do some customizations on the tuya_local/devices yaml files mainly to translate for my language. When i update to a newer version, theses customizations are lost on restart and i have to rewrite them. Is there a way to lock the yaml.

**Comment by make-all** [comment]
No there is not.  This is deliberate, to encourage people to contribute improvements rather than keep them locked up.

In this case, the proper fix is to implement the new translation capability that Home Assistant has for entity states.

**Comment by blamaz** [comment]
No clue on how to do this. Can you point me in a direction.

**Comment by make-all** [comment]
There is some documentation on the Home Assistant site https://developers.home-assistant.io/docs/internationalization/core#entities
I haven't been through it yet to see exactly what is involved and whether it will need code changes, or just extensions of the translation files.

Probably some review of the states used by different devices and making those consistent will help also.

**Comment by antonio1475** [comment]
@make-all while I totally understand your goal, can this be re-assessed or re-thought?

Example: I've worked on https://github.com/make-all/tuya-local/issues/585 and provided the config, but it hasn't been added to the last update. That's totally understandable and fine (priorities and all), but the file was deleted and the device couldn't be set-up.

Of course it's only one and it has an easy fix: re-upload the file and reload the device config in Tuya Local.

😄
